api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/surveillance/EligibleForIDSRFlagsCalculation.java

```diff
@@ -0,0 +1,729 @@
+				//Malaise
+				boolean patientMalaiseResultTriage = lastTriageEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastTriageEncounter, screeningQuestion, malaiseResult) : false;
+				boolean patientMalaiseResultGreenCard = lastHivFollowUpEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastHivFollowUpEncounter, screeningQuestion, malaiseResult) : false;
+				boolean patientMalaiseResultClinical = lastClinicalEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastClinicalEncounter, screeningQuestion, malaiseResult) : false;
+				//Weakness of limbs
+				boolean patientWeakLimbsResultTriage = lastTriageEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastTriageEncounter, screeningQuestion, limbsWeaknessResult) : false;
+				boolean patientWeakLimbsResultGreenCard = lastHivFollowUpEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastHivFollowUpEncounter, screeningQuestion, limbsWeaknessResult) : false;
+				boolean patientWeakLimbsResultClinical = lastClinicalEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastClinicalEncounter, screeningQuestion, limbsWeaknessResult) : false;
+
+				//Check admission status : Found in clinical encounter and type of visit
+				boolean patientAdmissionStatus = lastClinicalEncounter != null ? EmrUtils.encounterThatPassCodedAnswer(lastClinicalEncounter, adminQuestion, admissionAnswer) : false;
+				//Visit type is inpatient
+				Visit currentVisit = activeVisits.get(0);
+				Obs lastTempObs = EmrCalculationUtils.obsResultForPatient(tempMap, ptId);
+				if (lastTempObs != null) {
+					tempValue = lastTempObs.getValueNumeric();
+				}
+				//Triage
+				if (lastTriageEncounter != null) {
+					//1. SARI and ILI
+					if (patientFeverResultTriage && patientCoughResult) {
+						for (Obs obs : lastTriageEncounter.getObs()) {
+							dateCreated = obs.getDateCreated();
+							if (obs.getConcept().getConceptId().equals(DURATION)) {
+								duration = obs.getValueNumeric();
+							}
+							if (dateCreated != null) {
+								String createdDate = dateFormat.format(dateCreated);
+								if ((duration > 0.0 && duration < 10) && tempValue != null && tempValue >= 38.0) {
+									if (createdDate.equals(todayDate)) {
+										if (!patientAdmissionStatus && !currentVisit.getVisitType().getUuid().equals("a73e2ac6-263b-47fc-99fc-e0f2c09fc914")) {
+											eligible = true;
+											idsrMessage = idsrMessage.length() == 0 ? idsrMessage.append("ILI ") : idsrMessage.append(",ILI ");
+											break;
+										} else {
+											eligible = true;
+											idsrMessage = idsrMessage.length() == 0 ? idsrMessage.append("SARI ") : idsrMessage.append(",SARI ");											
+											break;
+										}
+									}
+								}
+							}							
+						}
+					}
+					//2. CHIKUNGUNYA
+					if (patientJointPainResultTriage && patientFeverResultTriage) {
```

Comment:
Let's have clear variable names i.e. triageEncounterHasJointPain or anything similar. Please check all the variables and update accordingly